### PR TITLE
Add docs on error handling in JS SDK

### PR DESCRIPTION
**Pull request by marikaner**
Add a few sentences on how to work with errors in the SDK.

### CODE REVIEW

File: docs/js/features/error-handling.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - error
+  - cause
+  - root cause
+---
+
+There are many reasons for errors to occur when developing software.
+The SAP Cloud SDK tries to provide as much information as possible and necessary to understand what caused an error.
+Generally speaking, there is no difference in handling errors when working with the SAP Cloud SDK.
+However, some errors are caused by underlying errors.
+In these cases the SAP Cloud SDK throws an `ErrorWithCause`, that allows access to the immediate cause and root cause of this error
```
Comment (jjtang1985) [inline docs/js/features/error-handling.mdx:20]:
```suggestion
In these cases, the SAP Cloud SDK throws an `ErrorWithCause`, that allows access to the immediate cause and root cause of this error.
```

File: docs/js/features/error-handling.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```ts
+import { BusinessPartner } from '@sap/cloud-sdk-vdm-business-partner-service';
+
+BusinessPartner.requestBuilder()
+  .create(BusinessPartner.builder().build())
+  .execute({ url: 'https://example.com' })
+  .catch(err => {
+    console.log('Error:', err.message);
+    console.log('Cause:', err.cause?.message);
+    console.log('Root cause:', err.rootCause?.message);
```
Comment (jjtang1985) [inline docs/js/features/error-handling.mdx:34]:
```suggestion
    console.log('Cause:', err.cause.message);
    console.log('Root cause:', err.rootCause.message);
```

**Review by jjtang1985 — APPROVED** [review]
Some minor suggestions.

File: docs/js/features/error-handling.mdx (hunk 2 above)
Comment (marikaner) [reply docs/js/features/error-handling.mdx:34]: This would fail if there was a pure error instead.

Comment (artemkovalyov) [reply docs/js/features/error-handling.mdx:34]: So the `cause` and `rootCause` might be missing?

**Review by artemkovalyov — APPROVED** [review]
Looks good

**Pull request merged**